csunplugged/topics/management/commands/_LessonsLoader.py

```diff
@@ -103,36 +130,68 @@ def load(self):
 
-            # Add programming exercises
+            # Add programming challenges
             if "programming-challenges" in lesson_structure:
-                programming_exercise_slugs = lesson_structure["programming-challenges"]
-                if programming_exercise_slugs is not None:
-                    for programming_exercise_slug in programming_exercise_slugs:
+                programming_challenge_slugs = lesson_structure["programming-challenges"]
+                if programming_challenge_slugs is not None:
+                    # Check all slugs are valid
+                    for programming_challenge_slug in programming_challenge_slugs:
                         try:
-                            programming_exercise = ProgrammingExercise.objects.get(
-                                slug=programming_exercise_slug,
+                            ProgrammingChallenge.objects.get(
+                                slug=programming_challenge_slug,
                                 topic=self.topic
                             )
-                            lesson.programming_exercises.add(programming_exercise)
+
                         except:
                             raise KeyNotFoundError(
                                 self.lessons_structure_file_path,
-                                programming_exercise_slug,
+                                programming_challenge_slug,
                                 "Programming Challenges"
                             )
 
+                    programming_challenges = ProgrammingChallenge.objects.filter(
+                        slug__in=programming_challenge_slugs,
+                        topic=self.topic
+                    ).order_by("challenge_set_number", "challenge_number")
+
+                    # Setup variables for counting
+                    display_set_number = 0
+                    last_set_number = -1
+                    display_number = 0
+                    last_number = -1
+
+                    for programming_challenge in programming_challenges:
+                        if programming_challenge.challenge_set_number > last_set_number:
+                            display_set_number += 1
+                            display_number = 0
+                            last_number = -1
+                        if programming_challenge.challenge_number > last_number:
+                            display_number += 1
+                        last_set_number = programming_challenge.challenge_set_number
+                        last_number = programming_challenge.challenge_number
+
+                        relationship = ProgrammingChallengeNumber(
+                            programming_challenge=programming_challenge,
+                            lesson=lesson,
+                            challenge_set_number=display_set_number,
+                            challenge_number=display_number,
+                        )
+                        relationship.save()
```

Comment:
this section needs comments. Usually less comments is fine if the code is fairly straight forward, but this is a little more involved so needs a couple comments to make it more clear.